### PR TITLE
Improve and test C backtraces

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: test/truffle/cexts/backtraces/expected.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	from /src/main/ruby/core/truffle/cext.rb:1209:in `block in rb_proc_new'
+	from /test/truffle/cexts/backtraces/bin/backtraces:8:in `call'
+	from /test/truffle/cexts/backtraces/bin/backtraces:8:in `bar'
+	from /src/main/ruby/core/truffle/cext.rb:1140:in `__send__'
+	from /src/main/ruby/core/truffle/cext.rb:1140:in `rb_funcall'
+	from backtraces.c:8:in `foo'
+	from /src/main/ruby/core/truffle/cext.rb:204:in `execute'
+	from /src/main/ruby/core/truffle/cext.rb:204:in `execute_with_mutex'
+	from /src/main/ruby/core/truffle/cext.rb:1385:in `foo'
+	from /test/truffle/cexts/backtraces/bin/backtraces:13:in `<main>'
```
Comment (eregon) [inline test/truffle/cexts/backtraces/expected.txt]: This is gonna be very fragile as it included line numbers of cext.rb.
They should be replaced somehow so adding a line in cext.rb does not require adapting this.

How is the prefix of paths removed BTW?
I don't see any usage of check from https://github.com/graalvm/truffleruby/blob/4731d7bf89f1ab37b1667eaea40291da31e57ce7/test/truffle/integration/backtraces/backtraces.rb#L9

Comment (eregon) [inline test/truffle/cexts/backtraces/expected.txt]: Ah, the gsub below is doing it.

Comment (eregon) [inline test/truffle/cexts/backtraces/expected.txt]: Then I would add
```ruby
gsub(/\/cext\.rb:(\d+)/, 'cext.rb:LINE')
```

Comment (eregon) [inline test/truffle/cexts/backtraces/expected.txt]: Alternatively, maybe all cext.rb frames should be hidden, as they are implementation details?

Comment (eregon) [inline test/truffle/cexts/backtraces/expected.txt]: The backtrace would look a lot cleaner without it at least.

Comment (chrisseaton) [inline test/truffle/cexts/backtraces/expected.txt]: Yes my next step was going to be optionally hiding C extension implementation frames. But I'll also fix the line number thing.

**Pull request merged**